### PR TITLE
KAFKA-12779: rename namedTopology in TaskId to topologyName

### PR DESCRIPTION
**Pull request by wcarlson5**
rename namedTopology is taks ID to topologyName to clear up confusion

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Review by ableegoldman — APPROVED** [review]
SGTM

**Review by mjsax — APPROVED** [review]
@kkonstantine This is a quick addendum to a KIP. As it's just renaming with zero risk, we want to cherry-pick to 3.0.

**Comment by kkonstantine** [comment]
Approved for 3.0. Thanks

File: streams/src/main/java/org/apache/kafka/streams/processor/TaskId.java [hunk 1]
```diff
@@ -76,13 +76,13 @@ public int partition() {
     /**
      * Experimental feature -- will return null
      */
-    public String namedTopology() {
-        return namedTopology;
+    public String topologyName() {
+        return topologyName;
```
Comment (izzyacademy) [inline streams/src/main/java/org/apache/kafka/streams/processor/TaskId.java:80]: I am just curious why this was not a getter (ie getTopologyName()) instead?

**Review by izzyacademy — COMMENTED** [review]
Overall changes look straightforward but I just had a question regarding why one of the methods on TaskId was not a real getter with the get* prefix.

File: streams/src/main/java/org/apache/kafka/streams/processor/TaskId.java (hunk 1 above)
Comment (wcarlson5) [reply streams/src/main/java/org/apache/kafka/streams/processor/TaskId.java:80]: It's by convention I think. If you look at all of the other "getters" none of them are prefixed with get either. I.e. partition() or subtopology()

**Comment by mjsax** [comment]
> Overall changes look straightforward but I just had a question regarding why one of the methods on TaskId was not a real getter with the get* prefix.

@izzyacademy It's custom in the whole Kafka code base, that we don't use the `getXxx` but `xxx()` for all getter methods.

**Comment by izzyacademy** [comment]
> > Overall changes look straightforward but I just had a question regarding why one of the methods on TaskId was not a real getter with the get* prefix.
> 
> @izzyacademy It's custom in the whole Kafka code base, that we don't use the `getXxx` but `xxx()` for all getter methods.

Thanks for the clarification @mjsax

**Review by izzyacademy — APPROVED** [review]
I have reviewed the changes and it looks good. 

Thanks for making the improvements.

**Comment by mjsax** [comment]
Merged to `trunk`. Did not cherry-pick cleanly to 3.0. Follow up PR for 3.0: https://github.com/apache/kafka/pull/11194

**Pull request merged**